File: packages/react-native-vision-camera-text-recognition/android/src/main/java/com/margelo/nitro/camera/textrecognition/HybridTextRecognizerFactory.kt
package com.margelo.nitro.camera.textrecognition

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip
import com.margelo.nitro.camera.HybridCameraOutputSpec

@DoNotStrip
@Keep
class HybridTextRecognizerFactory : HybridTextRecognizerFactorySpec() {
  @DoNotStrip
  @Keep
  override fun createTextRecognizer(): HybridTextRecognizerSpec {
    return HybridTextRecognizer()
  }

  @DoNotStrip
  @Keep
  override fun createTextRecognitionOutput(options: TextRecognitionOutputOptions): HybridCameraOutputSpec {
    return HybridTextRecognitionOutput(options)
  }
}

File: packages/react-native-vision-camera-text-recognition/ios/HybridTextRecognizerFactory.swift
//
//  HybridTextRecognizerFactory.swift
//  VisionCameraTextRecognition
//

import NitroModules
import VisionCamera

class HybridTextRecognizerFactory: HybridTextRecognizerFactorySpec {
  func createTextRecognizer() throws -> any HybridTextRecognizerSpec {
    return HybridTextRecognizer()
  }

  func createTextRecognitionOutput(options: TextRecognitionOutputOptions) throws
    -> any HybridCameraOutputSpec
  {
    return HybridTextRecognitionOutput(options: options)
  }
}
